File: src/main/java/ru/allformine/afmcp/commands/MessageCommand.java
package ru.allformine.afmcp.commands;

import org.spongepowered.api.command.CommandException;
import org.spongepowered.api.command.CommandResult;
import org.spongepowered.api.command.CommandSource;
import org.spongepowered.api.command.args.CommandContext;
import org.spongepowered.api.text.Text;
import org.spongepowered.api.text.format.TextColor;
import org.spongepowered.api.text.format.TextColors;
import ru.allformine.afmcp.Messaging;

import java.util.Arrays;

public class MessageCommand extends AFMCPCommand {
    @Override
    public CommandResult execute(CommandSource scr, CommandContext args) throws CommandException {
        if (!args.<String>getOne("type").isPresent() || !args.<String>getOne("message").isPresent()) {
            throw new CommandException(Text.of("Недостаточно аргументов!"));
        }

        String message = args.<String>getOne("message").get();
        Messaging.MessageType type;

        try {
            type = Messaging.MessageType.valueOf(args.<String>getOne("type").get().toUpperCase());
        } catch (IllegalArgumentException ignored) {
            reply(scr, Text.of("Неверный тип сообщения! Правильные типы:"));

            for (String messageType : Arrays.stream(Messaging.MessageType.values()).map(Enum::name).toArray(String[]::new)) {
                scr.sendMessage(Text.of(messageType));
            }

            return CommandResult.success();
        }

        Messaging.sendMessage(null, message, type);

        return CommandResult.success();
    }

    @Override
    public String getName() {
        return "Messaging";
    }

    @Override
    public TextColor getColor() {
        return TextColors.DARK_BLUE;
    }
}

File: src/main/java/ru/allformine/afmcp/commands/MessageCommand.kt
package ru.allformine.afmcp.commands

import org.spongepowered.api.command.CommandException
import org.spongepowered.api.command.CommandResult
import org.spongepowered.api.command.CommandSource
import org.spongepowered.api.command.args.CommandContext
import org.spongepowered.api.entity.living.player.Player
import org.spongepowered.api.text.Text
import org.spongepowered.api.text.format.TextColor
import org.spongepowered.api.text.format.TextColors
import ru.allformine.afmcp.Messaging

class MessageCommand : AFMCPCommand() {
    @Throws(CommandException::class)
    override fun execute(src: CommandSource, args: CommandContext): CommandResult {
        if (!args.getOne<String>("type").isPresent || !args.getOne<String>("message").isPresent) {
            throw CommandException(Text.of("Недостаточно аргументов!"))
        }

        val message = args.getOne<String>("message").get()

        if (message.length > 40) {
            reply(src, Text.of("Максимальная длина сообщения - 40 символов."))

            return CommandResult.success()
        }

        val type: Messaging.MessageType

        type = try {
            Messaging.MessageType.valueOf(args.getOne<String>("type").get().toUpperCase())
        } catch (ignored: IllegalArgumentException) {
            reply(src, Text.of("Неверный тип сообщения! Правильные типы:"))

            for (messageType in Messaging.MessageType.values()) {
                src.sendMessage(Text.of(messageType))
            }

            return CommandResult.success()
        }

        val player = args.getOne<Player>("player").orElse(null);

        Messaging.sendMessage(player, message, type)
        reply(src, Text.of("Сообщение было отправлено всем игрокам."))

        return CommandResult.success()
    }

    override fun getName(): String {
        return "Messaging"
    }

    override fun getColor(): TextColor {
        return TextColors.DARK_BLUE
    }
}